spec/models/borrowing_spec.rb

```diff
@@ -0,0 +1,85 @@
+require "rails_helper"
+
+RSpec.describe Borrowing, type: :model do
+  let(:user) {FactoryBot.create :user}
+  let(:category) {FactoryBot.create :category}
+  let(:author) {FactoryBot.create :author}
+  let(:publisher) {FactoryBot.create :publisher}
+
+  let!(:book) do
+    FactoryBot.create :book, category_id: category.id,
+                             publisher_id: publisher.id,
+                             author_id: author.id,
+                             quantity_borrowed: 8
+  end
+  let!(:borrowing_two) do
+    FactoryBot.create :borrowing, user_id: user.id,
+                                  borrow_code: "XATPVCZEBM",
+                                  status: :pending
+  end
+  let!(:borrowing_three) do
+    FactoryBot.create :borrowing, user_id: user.id,
+                                  borrow_code: "XATPVCZEBM",
+                                  status: :accept,
+                                  date_borrow: "2021-01-28 17:23:02",
+                                  date_pay: "2021-02-03 17:23:02"
+  end
+  let!(:borrow_item_first) do
+    FactoryBot.create :borrow_item, borrowing_id: borrowing_three.id,
+                                    book_id: book.id
+  end
+
+  describe "associations" do
+    it "belong to user" do
+      is_expected.to belong_to(:user).optional
+    end
+
+    it "has many borrow_items" do
+      is_expected.to have_many(:borrow_items).dependent :destroy
+    end
+
+    it "has many books" do
+      is_expected.to have_many(:books).through :borrow_items
+    end
+  end
+
+  describe "nested attributes" do
+    it "borrow_items" do
+      is_expected.to accept_nested_attributes_for :borrow_items
+    end
+  end
+
+  describe "enums" do
+    it "status" do
+      is_expected.to define_enum_for(:status)
+                 .with_values pending: 0, accept: 1, cancel: 2, payed: 3
+    end
+  end
+
+  describe ".return_quantity" do
+    it "return quantity book" do
+      borrowing_three.update status: 3
+      expect(borrowing_three.return_quantity).to eq [borrow_item_first]
+    end
+  end
+
+  describe ".accept_borrowing" do
+    it "return date borrow, date pay and send mail" do
+      expect(BorrowingMailer).to receive(:new_borrowing).and_call_original
+      borrowing_two.accept_borrowing
+    end
+  end
+
+  describe ".borrowing_cancel" do
+    it "return date borrow and date pay" do
+      expect(borrowing_two.borrowing_cancel.class).to eq ActionMailer::MailDeliveryJob
```

Comment:
mỗi 1 block it thì chỉ nên có 1 expect em nhé

Reply:
vâng ạ